db/dcrpg/pgblockchain.go

```diff
@@ -207,6 +213,7 @@ type ChainDB struct {
 	tpUpdatePermission map[dbtypes.TimeBasedGrouping]*trylock.Mutex
 	utxoCache          utxoStore
 	deployments        *ChainDeployments
+	piparser           *proposals.Parser
```

Comment:
Regarding my comments about architecture giving us the ability to change things out if needed, this is one place.  Instead of `*proposals.Parser`, this should really be an `interface`, such as `ProposalUpdater`.  e.g.

```go
type ProposalFetcher interface {
	ProposalsHistory() ([]*types.History, error)
	ProposalsHistorySince(since time.Time) ([]*types.History, error)
}
```

`github.com/dmigwi/go-piparser/proposals.Parser` satisfies this interface, but others can too and not require changes to the ChainDB type.